### PR TITLE
Add percent pruned before after winnery

### PR DESCRIPTION
**Pull request by dorimedini**
Results! ...are weird, I'm assuming initial weight sampling is not uniform in some interval? Different layers are given different-magnitude random initial weights?
Otherwise we should see a fixed percentage of each layer (the threshold percent)
@noamloya @galshachaf thoughts?

Addresses #97 

image
image

### CODE REVIEW

File: analyze_model.py [hunk 1]
```diff
@@ -53,11 +53,17 @@ def _rernd_layers(model, layers_indices):
...
+    def total_edges_in_layers(model, layers=[], count_nonzero_only=False):
         edges = 0
-        for layer in model.layers:
-            weights = layer.get_weights()
+        for layer in layers:
+            weights = model.layers[layer].get_weights()
             if weights:
                 edges += weights[0].size
+                if count_nonzero_only:
+                    edges -= np.count_nonzero(weights[0])
```
Comment (galshachaf) [inline analyze_model.py]: I think you have a bug here? if you are only counting non-zeroes, you should remove the zeros, not the non-zeroes, no?

**Review by galshachaf — CHANGES_REQUESTED** [review]

**Comment by galshachaf** [comment]
@dorimedini are you sure that you are returning the percentage? please not that the layers are different in size: the first layer is biggest (FC from 32*32 to h_dim), two middle ones are h_dim to h_dim, and the last one is smallest (h_dim to 10). Maybe you are not calculating the total number of edges properly?

File: analyze_model.py (hunk 1 above)
Comment (dorimedini) [reply analyze_model.py]: Looks like a bug, yes... wonder why the results look sane with this bug inside

Comment (dorimedini) [reply analyze_model.py]: Fixed

**Comment by dorimedini** [comment]
> @dorimedini are you sure that you are returning the percentage? please not that the layers are different in size: the first layer is biggest (FC from 32*32 to h_dim), two middle ones are h_dim to h_dim, and the last one is smallest (h_dim to 10). Maybe you are not calculating the total number of edges properly?

This explains why the results make sense... I returned the negation of the correct value in `total_edges_in_layers` and then computed the negation of what I meant when I wrote `pruned_percents.append(total_nonzero / total)` :D fixing...

**Comment by dorimedini** [comment]
> @dorimedini are you sure that you are returning the percentage? please not that the layers are different in size: the first layer is biggest (FC from 32*32 to h_dim), two middle ones are h_dim to h_dim, and the last one is smallest (h_dim to 10). Maybe you are not calculating the total number of edges properly?

Fixed

**Comment by galshachaf** [comment]
@dorimedini how are the new graphs looking?

**Review by galshachaf — APPROVED** [review]

**Comment by dorimedini** [comment]
> @dorimedini how are the new graphs looking?

@galshachaf with 30% prune threshold, these are the results. I'll open a task to show the accuracy of the trained+unpruned model vs. the trained+pruned model to make sure I'm not pruning too much / not enough

image
image
image
image

**Pull request merged**